src/Datadog.Trace/Span.cs

```diff
@@ -223,16 +244,11 @@ public Span SetTag(string key, string value)
                 default:
                     if (value == null)
                     {
-                        if (Tags != null)
-                        {
-                            // lock when modifying the collection
-                            lock (_tagsLock)
-                            {
-                                // Agent doesn't accept null tag values,
-                                // remove them instead
-                                Tags.Remove(key);
-                            }
-                        }
+                        RemoveTag(key);
+                    }
+                    else if (value.Value.IsMetrics)
```

Comment:
I think there's an unhandled edge case that we need to handle. Calling `SetTagValue("key1", 1); SetTagValue("key1", "non-number value")` will result in both `metrics["key1"]=1` and `tags["key1"]="non-number value"`, but we should only keep the second one

Reply:
You're totally right, I'm gonna make the change to handle that case.